tuxemon/event/actions/random_monster.py

```diff
@@ -0,0 +1,84 @@
+#
+# Tuxemon
+# Copyright (c) 2014-2017 William Edwards <shadowapex@gmail.com>,
+#                         Benjamin Bean <superman2k5@gmail.com>
+#
+# This file is part of Tuxemon
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+
+from __future__ import annotations
+
+import os
+import random as rd
+from typing import NamedTuple, Optional, Union, final
+
+from tuxemon import monster
+from tuxemon.event import get_npc
+from tuxemon.event.eventaction import EventAction
+from tuxemon.npc import NPC
+
+
+class RandomMonsterActionParameters(NamedTuple):
+    monster_level: int
+    trainer_slug: Union[str, None]
+
+
+@final
+class RandomMonsterAction(EventAction[RandomMonsterActionParameters]):
+    """
+    Add a monster to the specified trainer's party if there is room.
+
+    Script usage:
+        .. code-block::
+
+            random_monster <monster_level>[,trainer_slug]
+
+    Script parameters:
+        monster_level: Level of the added monster.
+        trainer_slug: Slug of the trainer that will receive the monster. It
+            defaults to the current player.
+
+    """
+
+    name = "random_monster"
+    param_class = RandomMonsterActionParameters
+
+    def start(self) -> None:
+
+        monster_level, trainer_slug = self.parameters
+
+        trainer: Optional[NPC]
+        if trainer_slug is None:
+            trainer = self.session.player
+        else:
+            trainer = get_npc(self.session, trainer_slug)
+
+        assert trainer, "No Trainer found with slug '{}'".format(
+            trainer_slug or "player"
+        )
+        
+        path = "mods/tuxemon/db/monster/"
+        
+        monsters = os.listdir(path)
```

Comment:
Can you get the monster from the DB object instead?

Reply:
I tried, but I was unable to get from the DB. Slug is necessary. Eventually we can insert temporary a list of monsters inside the file. Like this one:
_monsters = ["null", "rockitten", "rockat", "jemuar", "nut", "bolt", "arthrobolt", ..."]_

Reply:
Can't you get the slug from the DB?

Reply:
No, I can't, that's why I used os.listdir(path). I got the idea from locale.py, sure as you can see it utilizes fetch.
root = prepare.fetch("l18n")
for locale in os.listdir(root):

Reply:
from tuxemon.db import db

monster_slug = random.choice(db.database["monster"])

Reply:
I'd prefer this to having a hardcoded path.

Reply:
It was simpler than I thought, thanks, I'll update